nix/sources.nix

```diff
@@ -1,174 +1,26 @@
-# This file has been generated by Niv.
+{ system ? builtins.currentSystem }:
 
 let
+  lock = builtins.fromJSON (builtins.readFile ../flake.lock);
 
-  #
-  # The fetchers. fetch_<type> fetches specs of type <type>.
-  #
+  inputs = builtins.removeAttrs lock.nodes.${lock.root}.inputs [ "flake-utils" ];
 
-  fetch_file = pkgs: name: spec:
-    let
-      name' = sanitizeName name + "-src";
-    in
-      if spec.builtin or true then
-        builtins_fetchurl { inherit (spec) url sha256; name = name'; }
-      else
-        pkgs.fetchurl { inherit (spec) url sha256; name = name'; };
+  pkgs = let
+    input = lock.nodes.${inputs.nixpkgs}.locked;
+  in import (builtins.fetchTarball {
+    name = "source";
 
-  fetch_tarball = pkgs: name: spec:
-    let
-      name' = sanitizeName name + "-src";
-    in
-      if spec.builtin or true then
-        builtins_fetchTarball { name = name'; inherit (spec) url sha256; }
-      else
-        pkgs.fetchzip { name = name'; inherit (spec) url sha256; };
+    url = "https://github.com/${input.owner}/${input.repo}/archive/${input.rev}.tar.gz";
 
-  fetch_git = name: spec:
-    let
-      ref =
-        if spec ? ref then spec.ref else
-          if spec ? branch then "refs/heads/${spec.branch}" else
-            if spec ? tag then "refs/tags/${spec.tag}" else
-              abort "In git source '${name}': Please specify `ref`, `tag` or `branch`!";
-    in
-      builtins.fetchGit { url = spec.repo; inherit (spec) rev; inherit ref; };
+    sha256 = input.narHash;
+  }) { inherit system; };
 
-  fetch_local = spec: spec.path;
+in builtins.mapAttrs (input-name: node-name: let
```

Comment:
Worth a comment about why we use the builtin fetcher for nixpkgs but not for the rest.

Reply:
Fixed